### PR TITLE
Adding a (not so) basic users guide

### LINKED ISSUE #44: Where is the Documentation / Basic User Guide
Forgive me if I've overlooked something but there's no user guide or basic instructions on how to use it?
Not even a description of what each button does?

### PR DESCRIPTION
**Pull request by GameTec-live**
Im doing it, im adding a users guide to ideally adress as much as possible. (closes #44 )
And because apperently not a lot of people know how rfid works, etc, @nemanjan00 had the idea of creating a RFID wiki (unrelated, but worth mentioning)

### CODE REVIEW

**Comment by doegox** [comment]
I would suggest to use multiple pages in a subdir from the start as it will grow too much for a single file :)

**Comment by GameTec-live** [comment]
> I would suggest to use multiple pages in a subdir from the start as it will grow too much for a single file :)

just create a "documentation" folder then?

**Comment by GameTec-live** [comment]
> for the troubleshooting section:
> 
> * Python version mismatch, cf [TypeError: 'type' object is not subscriptable #90](https://github.com/RfidResearchGroup/ChameleonUltra/issues/90)
> * linux: permissions and modemmanager
> * LF: hardware issue related to screws, beware they are fragile...

will do!

**Comment by doegox** [comment]
> > for the troubleshooting section:
> > 
> > * Python version mismatch, cf [TypeError: 'type' object is not subscriptable #90](https://github.com/RfidResearchGroup/ChameleonUltra/issues/90)
> > * linux: permissions and modemmanager
> > * LF: hardware issue related to screws, beware they are fragile...
> 
> will do!

For python version mismatch, there is now a check in the CLI, now error should be clearer for users.

**Comment by doegox** [comment]
Beware many image links are broken (`C:\Users\bened\Desktop\ChameleonUltra\docs\images\lite.jpg` and the ultra images too).
Also better to use JPG for photos.

**Comment by GameTec-live** [comment]
> Beware many image links are broken (`C:\Users\bened\Desktop\ChameleonUltra\docs\images\...` and the ultra images too). Also better to use JPG for photos.

should be fixed...

**Comment by doegox** [comment]
> > Beware many image links are broken (`C:\Users\bened\Desktop\ChameleonUltra\docs\images\...` and the ultra images too). Also better to use JPG for photos.
> 
> should be fixed...

not until there are new commits ;)

**Comment by doegox** [comment]
ok I'll push comments/suggestions on the hw file, should be easy for you to integrate

File: docs/hardware.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+The Chameleon Ultra comes in a black box with gold printing. This box has the dimensions: 9.5cm x 5.5cm x 3.5cm
+
+![box.png](./images/ultra-box.jpg)
+
+The Box contains a foam pad, a USB cable that has a removable end to convert it to USB-C, a Proxgrind 3.5 hex screwdriver, 2 replacement screws and a keychain and the device itself.
+
+![box-contents.jpg](./images/ultra-box-contents.jpg)
+
+The deivce itself features 4 screws holding it together, 2 Buttons labled `A` and `B`.  The Device consists of 2 PCBs (**P**rinted **C**ircuit **B**oards) and a plastic spacer, one contains the Electronic and the HF (**H**igh **F**requency), 13.56 MHz, antenna as well as the 8 Leds indecating which is currently active and the other board contains the Chameleon Ultra Text, the screws and the LF (**L**ow **F**requency), 125KHz, antenna. The plastic spacer houses the battery as well as the ferrite pad which enables HF and LF emulation at the same time. It also has has the USB-C charging and data port and a hole for inserting the keychain loop. The Chameleon Ultras dimensions are: 2.4cm x 4cm x 8mm
```
Comment (doegox) [inline docs/hardware.md:15]:
```suggestion
The device itself features 4 screws holding it together, 2 buttons labeled `A` and `B`.  The device consists of 2 PCBs (**P**rinted **C**ircuit **B**oards) and a plastic spacer, one contains the Electronic and the HF (**H**igh **F**requency), 13.56 MHz, antenna as well as the 8 LEDs indicating which slot is currently active and the other board features the Chameleon Ultra text, the screws and the LF (**L**ow **F**requency), 125KHz, antenna. The plastic spacer houses the battery as well as the ferrite pad which enables HF and LF emulation at the same time. It also has has the USB-C charging and data port and a hole for inserting the keychain loop. The Chameleon Ultra dimensions are: 2.4cm x 4cm x 8mm 
```

File: docs/hardware.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+![box-contents.jpg](./images/ultra-box-contents.jpg)
+
+The deivce itself features 4 screws holding it together, 2 Buttons labled `A` and `B`.  The Device consists of 2 PCBs (**P**rinted **C**ircuit **B**oards) and a plastic spacer, one contains the Electronic and the HF (**H**igh **F**requency), 13.56 MHz, antenna as well as the 8 Leds indecating which is currently active and the other board contains the Chameleon Ultra Text, the screws and the LF (**L**ow **F**requency), 125KHz, antenna. The plastic spacer houses the battery as well as the ferrite pad which enables HF and LF emulation at the same time. It also has has the USB-C charging and data port and a hole for inserting the keychain loop. The Chameleon Ultras dimensions are: 2.4cm x 4cm x 8mm 
+
+![ultra.png](./images/ultra.jpg)
+
+## The Lite
+
+The Chameleon Lite comes in a white box with blue printing. This box has the dimensons: 9.5cm x 6cm x 3.5cm
```
Comment (doegox) [inline docs/hardware.md:21]:
```suggestion
The Chameleon Lite comes in a white box with blue printing. This box has the following dimensions: 9.5 cm x 6 cm x 3.5 cm
```

File: docs/hardware.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Hardware
+
+The Chameleon comes in 3 Hardware variants, the Ultra, the Lite and the Devkit.
+
+## The Ultra
+
+The Chameleon Ultra comes in a black box with gold printing. This box has the dimensions: 9.5cm x 5.5cm x 3.5cm
```
Comment (doegox) [inline docs/hardware.md:7]:
```suggestion
The Chameleon Ultra comes in a black box with gold printing. This box has the following dimensions: 9.5 cm x 5.5 cm x 3.5 cm
```

**Comment by GameTec-live** [comment]
Id say that firmware is also mostly done? It feels like something is still missing, but idk what? (For compiling, etc, theres the "How_to_use_firmware.md", ig i could merge that into there too if needed?)

File: docs/hardware.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+![lite-box.-contents.jpg](./images/lite-box-contents.jpg)
+
+The device itself features 2 Buttons labled with arrows. The Device consists of one PCB in a blue plastic housing. This one PCB contains the Electronic and the HF, antenna as well as the 8 Leds indecating which is currently active and the USB-C port. The LF antenna is glued onto the back of the PCB and is visible through the housing. The Battery is soldered in place and the housing is hold together by thin fragile pins which are easy to snap, it is not designed to be disassembled. The Keychain Loop is also relatively fragile so be careful. The Chameleon Lites dimensions are: 3.6cm x 6.1cm x 8mm
+
+![lite.jpg](./images/lite.jpg)
+
+## The Devkit
+
+Just like the Chameleon Ultra, the Devkit comes in a black box with gold printing. This box has the dimensions: 12cm x  8cm x 3.5cm
```
Comment (doegox) [inline docs/hardware.md:35]:
```suggestion
Just like the Chameleon Ultra, the Devkit comes in a black box with gold printing. This box has the following dimensions: 12 cm x  8 cm x 3.5 cm
```

File: docs/hardware.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+Again, just like the lite, the box contains a foam pad, a USB cable that has a removable end to convert it to USB-C and the device itself.
+
+![devkit-box-contents.jpg](./images/devkit-box-contents.jpg)
+
+The deivce itself features 2 Buttons labled `A` and `B`. The Device is only one PCB without a case. At the bottom of this PCB both the HF and LF coil are found. Because its a Devkit, this Chameleon has its SWD (**S**ingle **W**ire **D**ebug) port and some testpoints exposed. (In the photos below a pinheader is already soldered into the SWD port, this is not the case from factory) The Chameleon Devkits dimensions are:  5.3cm x 8.5cm x 5mm (with rubber feet, 1.5mm without)
+
+![devkit.jpg](./images/devkit.jpg)
+
+## Whats the difference?
```
Comment (doegox) [inline docs/hardware.md:47]:
```suggestion
## What is the difference between the Lite and the Ultra/DevKit?
```

File: docs/home.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+This Guide is split up into multiple "subguides":
+
+- ["Hardware"](./hardware.md) Learn to know the hardware of your Chameleon
+
+- ["Firmware"](./firmware.md) Your Chameleon runs a firmware, learn what it can do and how to use it
+
+- ["CLI"](./cli.md) The official way to control your Chameleon is via the **C**ommand **L**ine **I**nterface (CLI) . Learn how to install and master the CLI.
+
+- ["GUIs"](./gui.md) Some people also develop **G**raphical **U**ser **I**nterfaces (GUIs), theese may be a good start for people that dont want to deal with a CLI.
```
Comment (doegox) [inline docs/home.md:13]:
```suggestion
- ["GUIs"](./gui.md) Some people also develop **G**raphical **U**ser **I**nterfaces (GUIs), theese may be a good start for people that do not want to deal with a CLI.
```

**Comment by doegox** [comment]
> Id say that firmware is also mostly done? It feels like something is still missing, but idk what? (For compiling, etc, theres the "How_to_use_firmware.md", ig i could merge that into there too if needed?)

I would separate dev things and fw things for users. So you can add a subsection "development.md" which will basically be the current How_to_use_Firmware.md
And in the firmware.md I see topics for users, like:
* how to get precompiled fw and how to flash it
* what buttons and LEDs do mean in the various usages?
  * B on boot = DFU
  * A/B on app -> configurable, default behaviors are slot +1/-1 & long = copy IC/ID
  * slot selects, red=.. green=... blue=...
  * 4 pulsing green when charging (I'm not sure, <4 when battery is low?)
  * when CLI connected, LED chase in color of selected slot, left to right if slot 1,2,3,4, right to left if slot 5,6,7,8
  * white LED for external field detection (if current slot supports corresponding HF/LF)
  * sleep after N secs, can be interrupted by pressing a button
 And I'm probably forgetting things. I'm not fully aware of all the possible animations and their meaning

**Comment by doegox** [comment]
When you thing a section or 2 are ok, we can just merge this PR and open a new PR so we don't wait for all the guide to be ready before showing parts

File: docs/hardware.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+The Chameleon Lite comes in a white box with blue printing. This box has the dimensons: 9.5cm x 6cm x 3.5cm
+
+![lite-box.jpg](./images/lite-box.jpg)
+
+The Box contains a foam pad, a USB cable that has a removable end to convert it to USB-C and the device itself.
+
+![lite-box.-contents.jpg](./images/lite-box-contents.jpg)
+
+The device itself features 2 Buttons labled with arrows. The Device consists of one PCB in a blue plastic housing. This one PCB contains the Electronic and the HF, antenna as well as the 8 Leds indecating which is currently active and the USB-C port. The LF antenna is glued onto the back of the PCB and is visible through the housing. The Battery is soldered in place and the housing is hold together by thin fragile pins which are easy to snap, it is not designed to be disassembled. The Keychain Loop is also relatively fragile so be careful. The Chameleon Lites dimensions are: 3.6cm x 6.1cm x 8mm
```
[the remark this answers is not in the document]
Comment (GameTec-live) [reply docs/hardware.md:29]: why are here floating ``` ?

[the remark this answers is not in the document]
Comment (doegox) [reply docs/hardware.md]: my mistake, they should have been on the next line to finish the "suggestion" block. Fixed now

File: docs/hardware.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+Just like the Chameleon Ultra, the Devkit comes in a black box with gold printing. This box has the dimensions: 12cm x  8cm x 3.5cm
+
+![devkit-box.jpg](./images/devkit-box.jpg)
+
+Again, just like the lite, the box contains a foam pad, a USB cable that has a removable end to convert it to USB-C and the device itself.
+
+![devkit-box-contents.jpg](./images/devkit-box-contents.jpg)
+
+The deivce itself features 2 Buttons labled `A` and `B`. The Device is only one PCB without a case. At the bottom of this PCB both the HF and LF coil are found. Because its a Devkit, this Chameleon has its SWD (**S**ingle **W**ire **D**ebug) port and some testpoints exposed. (In the photos below a pinheader is already soldered into the SWD port, this is not the case from factory) The Chameleon Devkits dimensions are:  5.3cm x 8.5cm x 5mm (with rubber feet, 1.5mm without)
```
[the remark this answers is not in the document]
Comment (GameTec-live) [reply docs/hardware.md]: fixed

File: docs/firmware.md [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 The Chameleons firmware consists of 3 parts, the bootloader, the application and the settings
 
-NOTE: If your developer searching for the building instructions, look into [How_to_use_Firmware.md](../How_to_use_Firmware.md)
+NOTE: If your developer searching for the building instructions, look into [development](./development)
```
Comment (doegox) [inline docs/firmware.md:5]:
```suggestion
NOTE: If your developer searching for the building instructions, look into [development](./development.md)
```

**Comment by doegox** [comment]
oops I missed when you committed the first half of firmware.md so I can't add suggestions anymore on that part. 
I will push directly the changes, only to firmware.md, in this branch

**Comment by doegox** [comment]
ok I've pushed more than firmware.md changes :D
for CLI: I think it's better to write a cheat sheet with common scenarios chaining the proper commands than an exhaustive list of commands that will be painful to maintain and that users can easily discover by themselves in the client,

**Comment by ComicBit** [comment]
Would be good to make a timeline of the future developments, I can help

**Comment by doegox** [comment]
Well this is opensource, everyone is contributing to what he prefers whenever he finds time.
I'm not sure how you could plan a roadmap with a timeline...

**Comment by ComicBit** [comment]
@doegox I am sorry but I don't know if there any paid contributors to the project. Of course there is no way to plan at a fast pace in an open source project if everyone is indeed doing it in their free time. Probably the right word I needed to use would have been Roadmap, to functionalities etc, and probably with large tolerance regarding the time. Just to be a conceptual map of what comes after and if someone wants to join, what they can indeed do to contribute

**Comment by doegox** [comment]
Yes if you can initiate sth similar to https://github.com/RfidResearchGroup/proxmark3/wiki/Public-Roadmap that would be great!
Similarly, I think 1 or 2 wiki pages would suffice (roadmap & wishlist).

**Comment by ComicBit** [comment]
I would love that, can you tell me how to do it tho? I think I don't know where to take the accomplished parts already, and regarding the future I guess it would require a meeting of people understanding what would be the next things in the line and plan a draw for it. And whoever will feel like developing will do.

**Comment by doegox** [comment]
@ComicBit a good start is to collect the things still to be implemented detailed in https://github.com/RfidResearchGroup/ChameleonUltra/blob/main/docs/technical_whitepaper.md and to collect feature ideas expressed in the open issues https://github.com/RfidResearchGroup/ChameleonUltra/issues

**Comment by ComicBit** [comment]
@doegox I opened a PR for the docs, give it a look, the things in the wishlist are there just as placeholder. I thought we could make a wiki already considering the amount of documents.

**Comment by wzqvip** [comment]
seems there's still no "key function" explained in detail currently. I wonder what's "charge" does. The option did not appear in https://docs.chameleonultragui.dev/.

**Comment by GameTec-live** [comment]
> seems there's still no "key function" explained in detail currently. I wonder what's "charge" does. The option did not appear in https://docs.chameleonultragui.dev/.

it simply displays the battery charge on the leds... I should prob add it to the docs at some point...

**Pull request merged**